marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/entitySearch/getOpticPlan.sjs

```diff
@@ -0,0 +1,79 @@
+/*
+  Copyright (c) 2020 MarkLogic Corporation
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+'use strict';
+
+const op = require('/MarkLogic/optic');
+const search = require('/MarkLogic/appservices/search/search');
+
+const returnFlags = `<return-aggregates xmlns="http://marklogic.com/appservices/search">false</return-aggregates>
+  <return-constraints xmlns="http://marklogic.com/appservices/search">false</return-constraints>
+  <return-facets xmlns="http://marklogic.com/appservices/search">false</return-facets>
+  <return-frequencies xmlns="http://marklogic.com/appservices/search">false</return-frequencies>
+  <return-metrics xmlns="http://marklogic.com/appservices/search">false</return-metrics>
+  <return-plan xmlns="http://marklogic.com/appservices/search">false</return-plan>
+  <return-qtext xmlns="http://marklogic.com/appservices/search">false</return-qtext>
+  <return-results xmlns="http://marklogic.com/appservices/search">false</return-results>
+  <return-similar xmlns="http://marklogic.com/appservices/search">false</return-similar>
+  <return-values xmlns="http://marklogic.com/appservices/search">false</return-values>
+  <return-query xmlns="http://marklogic.com/appservices/search">true</return-query>`;
+
+const stylesheet = fn.head(xdmp.unquote(`<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="1.0">
+   <xsl:template match="node()|@*">
+      <xsl:copy>
+         <xsl:apply-templates select="node()|@*" />
+      </xsl:copy>
+   </xsl:template>
+   <xsl:template match="*:return-aggregates|*:return-constraints|*:return-facets|*:return-frequencies|*:return-metrics|*:return-plan|*:return-qtext|*:return-results|*:return-similar|*:return-values|*:return-query" />
+   <xsl:template match="*:options">
+      <xsl:copy>
+         <xsl:apply-templates select="node()|@*" />
+         ${returnFlags}
+      </xsl:copy>
+   </xsl:template>
+</xsl:stylesheet>`));
+
+
+var viewName;
+var schemaName;
+var limit;
+var structQuery;
+var searchQText;
+var queryOptions;
```

Comment:
I think you can define the DS API with queryOptions as type = jsonDocument, and then pass (hopefully) a JacksonHandle into it in the Java code. Same goes for the structQuery - if these are JSON objects, let's have a richer interface than using "String".

Also, "searchText" is widely used as the parameter name for a user's search text, so let's use that instead of "searchQText". 

One other nit - generally, I'm opposed to abbreviating words unless there's a really, really good reason. So let's say "structuredQuery" instead of "structQuery".

Reply:
Both of them are XML docs which I get as a String from the relevant methods in the middle tier.
Converting them to an Object would mean a lot of unnecessary serialization/deserialization at both ends of the wire.